File: backend/app/schemas/remote_workspace.py
# SPDX-FileCopyrightText: 2026 Weibo, Inc.
#
# SPDX-License-Identifier: Apache-2.0

from typing import List, Optional

from pydantic import BaseModel


class RemoteWorkspaceStatusResponse(BaseModel):
    connected: bool
    available: bool
    root_path: str = "/workspace"
    reason: Optional[str] = None


class RemoteWorkspaceTreeEntry(BaseModel):
    name: str
    path: str
    is_directory: bool
    size: int = 0
    modified_at: Optional[str] = None


class RemoteWorkspaceTreeResponse(BaseModel):
    path: str
    entries: List[RemoteWorkspaceTreeEntry]

File: frontend/src/apis/remoteWorkspace.ts
// SPDX-FileCopyrightText: 2026 Weibo, Inc.
//
// SPDX-License-Identifier: Apache-2.0

import { apiClient } from './client'

export type RemoteWorkspaceDisposition = 'inline' | 'attachment'

export interface RemoteWorkspaceStatusResponse {
  connected: boolean
  available: boolean
  root_path: string
  reason: string | null
}

export interface RemoteWorkspaceTreeEntry {
  name: string
  path: string
  is_directory: boolean
  size: number
  modified_at?: string | null
}

export interface RemoteWorkspaceTreeResponse {
  path: string
  entries: RemoteWorkspaceTreeEntry[]
}

export const remoteWorkspaceApis = {
  getStatus(taskId: number) {
    return apiClient.get<RemoteWorkspaceStatusResponse>(`/tasks/${taskId}/remote-workspace/status`)
  },

  getTree(taskId: number, path: string) {
    return apiClient.get<RemoteWorkspaceTreeResponse>(
      `/tasks/${taskId}/remote-workspace/tree?path=${encodeURIComponent(path)}`
    )
  },

  getFileUrl(taskId: number, path: string, disposition: RemoteWorkspaceDisposition) {
    return `/api/tasks/${taskId}/remote-workspace/file?path=${encodeURIComponent(path)}&disposition=${disposition}`
  },
}
